geode-core/src/main/java/org/apache/geode/cache/client/PoolFactory.java

```diff
@@ -73,6 +73,14 @@
    */
   int DEFAULT_FREE_CONNECTION_TIMEOUT = 10000;
 
+  /**
+   * The default amount of time, in milliseconds, which we will wait for a server connection if max
+   * connections is set and there is no free connections towards designated server.
+   * <p>
+   * Current value: <code>-1</code>.
+   */
+  int DEFAULT_SERVER_CONNECTION_TIMEOUT = -1;
```

Comment:
A timeout of -1 is usually used to imply that the timeout is infinite, i.e. that we will wait forever to obtain a server connection if max connections is set and there is no free connections towards designated server. Would it be possible to change the default value to 0 but still ensure that the original behaviour is preserved?

Reply:
For timeout, if you look at the code (checkShutdownInterruptedOrTimeout) -1 means no timeout, because you compare current time against, current + timeout. If we set 0, you will get the code to retry several times until 1ms passes.

Reply:
If the comparison in `checkShutdownInterruptedOrTimeout` was `timeout <= System.nanoTime()` then this problem would be solved, I think.

Reply:
OK. Updated.